index.js

```diff
@@ -90,8 +93,49 @@ function handleClose (writeStream) {
       writeStream.emit('close')
     })
   }
+  function trapWindowsEPERMRename (err) {
+    if (err.syscall && err.syscall === 'rename' &&
+        err.code && err.code === 'EPERM' &&
+        writeStream.__isWin
+    ) {
+      if (fs.existsSync(writeStream.__atomicTarget)) {
+        var tmpHash = crypto.createHash('sha256')
+        var targetHash = crypto.createHash('sha256')
+
+        fs.readFile(writeStream.__atomicTmp,
```

Comment:
Hi @iarna, thank you for the feedback. I made some changes to fix the issues you pointed out, but I'm not quite sure about this one

> The only slightly jinky bit here is that on error you want to call cleanup() with the top-level error, so just passing cleanup to iferr wouldn't work. But all that calls for is wrapping cleanup with something that does that.

Could you elaborate on this so I can implement this as well?